public/app/features/explore/TraceView/components/utils/color-generator.tsx

```diff
@@ -132,19 +132,20 @@ export function getRgbColorByKey(key: string, theme: GrafanaTheme2): [number, nu
 }
 
 export function getFilteredColors(colorsHex: string[], theme: GrafanaTheme2) {
+  const filtered = [...colorsHex];
```

Comment:
@joey-grafana fyi did not end up using this for now but I thought it make sense to keep this fix here. As we just passed the colors from packages/grafana-ui/src/utils/colors.ts in the generator, this was modifying that global array which is used in other places which could make other colors glitch as other parts just picked colors by index.